packages/@uppy/drop-target/src/index.ts

```diff
@@ -1,24 +1,47 @@
+import type { Body, Meta, UppyFile } from '@uppy/utils/lib/UppyFile'
+import type { Uppy, State } from '@uppy/core/src/Uppy.ts'
+import type { UIPluginOptions } from '@uppy/core/src/UIPlugin.ts'
 import BasePlugin from '@uppy/core/lib/BasePlugin.js'
 import getDroppedFiles from '@uppy/utils/lib/getDroppedFiles'
 import toArray from '@uppy/utils/lib/toArray'
-
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
 
-function isFileTransfer (event) {
-  return event.dataTransfer.types?.some((type) => type === 'Files') ?? false
+interface DropTargetOptions extends UIPluginOptions {
+  target?: HTMLElement | string | null
+  onDrop?: (event: DragEvent) => void
+  onDragOver?: (event: DragEvent) => void
+  onDragLeave?: (event: DragEvent) => void
+}
+
+interface DragEventWithFileTransfer extends DragEvent {
+  dataTransfer: NonNullable<DragEvent['dataTransfer']>
+}
+
+function isFileTransfer(event: DragEvent): event is DragEventWithFileTransfer {
+  return event.dataTransfer?.types?.some((type) => type === 'Files') ?? false
```

Comment:
Do we need `DragEventWithFileTransfer` to make `dataTransfer` `NonNullable`? It's probably nullable for a reason as it can happen in browsers. Maybe just an extra check in code instead of ignore the possibility?

Reply:
But when we call `isFileTransfer` we only get the item that has non-nullish `dataTransfer`, this was the logic behind the event, right @aduh95?

Reply:
You're looking at the extra check in the code, we are _not_ ignoring the possiblity.

Reply:
What about adding one more `?.` and you don't need `DragEventWithFileTransfer`?

Reply:
Why don't you try it yourself and report back? I can bet you'll find the current implementation is the correct one.